### ISSUE TITLE
Only CUDA 0 being used, after using torch.cuda.is_available() method

### ISSUE THREAD

**Issue by willook** [ultralytics/ultralytics#19084]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

When I use the torch.cuda.is_available() method, the YOLO model always uses only GPU 0, regardless of the CUDA device number I assign.

### Environment

Ultralytics 8.3.59 🚀 Python-3.10.12 torch-2.3.0a0+40ec155e58.nv24.03 CUDA:0 (NVIDIA RTX A6000, 48670MiB)
Setup complete ✅ (64 CPUs, 251.6 GB RAM, 1803.9/3666.0 GB disk)

OS                  Linux-6.8.0-51-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
RAM                 251.55 GB
Disk                1803.9/3666.0 GB
CPU                 AMD EPYC 7513 32-Core Processor
CPU count           64
GPU                 NVIDIA RTX A6000, 48670MiB
GPU count           4
CUDA                12.4

numpy               ✅ 1.24.4>=1.23.0
numpy               ✅ 1.24.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.8.3>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.12.0>=1.4.1
torch               ✅ 2.3.0a0+40ec155e58.nv24.3>=1.8.0
torch               ✅ 2.3.0a0+40ec155e58.nv24.3!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.18.0a0>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 5.9.4
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 1.5.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0
{'CPU': 'AMD EPYC 7513 32-Core Processor',
 'CPU count': 64,
 'CUDA': '12.4',
 'Disk': '1803.9/3666.0 GB',
 'Environment': 'Linux',
 'GPU': 'NVIDIA RTX A6000, 48670MiB',
 'GPU count': 4,
 'Install': 'pip',
 'OS': 'Linux-6.8.0-51-generic-x86_64-with-glibc2.35',
 'Package Info': {'matplotlib': '✅ 3.8.3>=3.3.0',
                  'numpy': '✅ 1.24.4<2.0.0; sys_platform == "darwin"',
                  'opencv-python': '✅ 4.10.0.84>=4.6.0',
                  'pandas': '✅ 1.5.3>=1.1.4',
                  'pillow': '✅ 10.2.0>=7.1.2',
                  'psutil': '✅ 5.9.4',
                  'py-cpuinfo': '✅ 9.0.0',
                  'pyyaml': '✅ 6.0.1>=5.3.1',
                  'requests': '✅ 2.31.0>=2.23.0',
                  'scipy': '✅ 1.12.0>=1.4.1',
                  'seaborn': '✅ 0.13.2>=0.11.0',
                  'torch': '✅ 2.3.0a0+40ec155e58.nv24.3!=2.4.0,>=1.8.0; '
                           'sys_platform == "win32"',
                  'torchvision': '✅ 0.18.0a0>=0.9.0',
                  'tqdm': '✅ 4.67.1>=4.64.0',
                  'ultralytics-thop': '✅ 2.0.13>=2.0.0'},
 'Python': '3.10.12',
 'RAM': '251.55 GB'}

### Minimal Reproducible Example

When I run the first code, it uses GPU 3, whereas the second code uses GPU 0.

```
# use GPU 3
from ultralytics import YOLO

model = YOLO("yolo11m-pose", task="pose")
model.predict(video_file, device="3")
```

```
# use GPU 0
import torch
from ultralytics import YOLO

torch.cuda.is_available()
model = YOLO("yolo11m-pose", task="pose")
model.predict(video_file, device="3")
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @willook, thank you for reporting this issue and for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for guidance, where you’ll find many helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that may provide insights into common issues.

If this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to better help us investigate the behavior. It appears you've provided some code snippets already—thank you! Please ensure these clearly reproduce the issue as described, and confirm the difference in device usage between the two snippets.

In the meantime, here are some general steps to ensure your environment is set up correctly:
1. Upgrade to the latest version of the `ultralytics` package as well as all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

2. Verify that the proper GPU properties are being recognized and utilized, and ensure your code aligns with our [multigpu usage tips](https://docs.ultralytics.com/modes/val/).

For running YOLO in verified environments preconfigured with dependencies, you can try:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For real-time support, consider joining our community:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for quick chat and team access
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for sharing knowledge with the wider community

Finally, we’d like to assure you that this is an automated response to get things started 😊. An Ultralytics engineer will take a closer look at your issue and provide additional assistance shortly. Happy coding! 🚀

**Comment by Y-T-G** [comment]
You can try using `device=torch.device("cuda:3")`
